gems/pending/util/duplicate_blocker/dedup_handler.rb

```diff
@@ -63,8 +68,8 @@ def handle(method, *args)
       key = key_generator.call(method, *args)
       desc = descriptor.call(method, *args)
 
-      entry = update_history(key, desc)
-      tripped?(entry) ? on_blocking_call(desc) : method[*args]
+      entry = update_history(key.hash, desc)
+      tripped?(entry, desc) ? on_blocking_call(desc) : method[*args]
```

Comment:
It seems like the entry and the description are always coupled together, which is why you have to keep passing it along to every method.  Since you generate the entry from the update_history method and pass in the `desc` anyway...maybe the desc should just be a property *of* the entry.  Then, you don't have to pass it everywhere, and when you need it you just call `entry.description`